src/main/java/com/exasol/adapter/document/documentfetcher/files/JsonLinesIterator.java

```diff
@@ -18,11 +19,14 @@
 class JsonLinesIterator implements Iterator<DocumentNode<JsonNodeVisitor>> {
     private final BufferedReader jsonlReader;
     private final InputStream jsonlStream;
+    private final FileLoader fileLoader;
     private final InputStreamReader inputStreamReader;
     private String nextLine = null;
+    private long lineCounter = 0;
 
     JsonLinesIterator(final FileLoader fileLoader) {
         this.jsonlStream = fileLoader.loadFile();
+        this.fileLoader = fileLoader;
```

Comment:
So far, it is only needed for filename. Only get the fileName from fileLoader parameter.